nym-vpn-x/src-tauri/src/fs/path.rs

```diff
@@ -1,24 +1,71 @@
-use once_cell::sync::Lazy;
-use std::path::PathBuf;
-use tauri::api::path::cache_dir;
-use tracing::error;
-
 use crate::fs::util::check_dir;
 use crate::APP_DIR;
+use once_cell::sync::Lazy;
+use std::path::PathBuf;
+use tracing::{debug, error, info};
 
 pub const LOG_DIR: &str = "log";
 
-pub static LOG_DIR_PATH: Lazy<Option<PathBuf>> = Lazy::new(|| {
-    let log_dir = cache_dir()
-        .map(|mut p| {
-            p.push(APP_DIR);
-            p.push(LOG_DIR);
-            p
-        })
-        .and_then(|p| check_dir(&p).map(|_| p).ok());
+/// The app config directory
+/// - Linux: `$XDG_CONFIG_HOME/nymvpn-x` or `$HOME/.config/nymvpn-x`
+/// - Windows: `{FOLDERID_RoamingAppData}\nymvpn-x`, example  `C:\Users\Pierre\AppData\Roaming\nymvpn-x`
+pub static APP_CONFIG_DIR: Lazy<Option<PathBuf>> = Lazy::new(|| {
```

Comment:
Typically you'd access the same info from tauri using `App` instance:

```rs
let config_dir = app.path().app_config_dir()?;
```

Maybe there is a reason why this is static, but if you can obtain these paths from tauri and then push them down to dependencies, you may be able to get rid of dependency on `dirs` crate.

https://docs.rs/tauri/2.0.0-rc.11/tauri/path/struct.PathResolver.html

Reply:
Yes I know. 
In short, I need to access these dirs before tauri app is instantiated for a bunch of initialization which happen before tauri app setup.

And so this is why I'm using dirs crate to resolve them (note that underthewood this is what tauri is using too, so it's the same https://github.com/tauri-apps/tauri/discussions/7757).
Also note that, in tauri v1, the dirs API was exposed outside the app handle. This is not the case anymore. Reason why I had to add `dirs` in deps.

Reply:
This is an interesting one. I think it touches on a broader thing where it would be pretty great if actually this can be swapped around. Bringing up the UI before reading config files and stuff like, so that we can present errors related to this in the UI. I saw the startup_error window and I think that's a good start, and we should capture all errors in the UI

That would be out of scope of this PR though.

Reply:
We spoke about that over the call. Basically if we move initialization into Tauri's `setup(|app| { ... })` then we can access paths directly from tauri's facilities. @doums said he will consider a refactor down the road.

Reply:
Yea actually, and thanks to Andrej for pointing it; there is a way to initiate those states in the tauri app setup. In the required non-optional way.
That would allow dropping that dedicated app 'startup_error' and handle all startup failure directly in the main app. But as you said, this is out of the scope here. But good to keep in mind in future iteration.